Mathlib/Topology/ApproximateUnit.lean

```diff
@@ -3,146 +3,80 @@ Copyright (c) 2024 Jireh Loreaux. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Jireh Loreaux
 -/
-import Mathlib.Topology.Bornology.Basic
-import Mathlib.Topology.Basic
+import Mathlib.Topology.MetricSpace.Bounded
+import Mathlib.Topology.Algebra.Monoid
 
 /-! # Approximate units
 
-An *approximate unit* is a filter basis such that multiplication on the left (or right) by `m`
-tends to `m` along the filter generated by the basis. In the case of normed rings, it is often
-required that the elements of the basis are contained in the unit ball. In order to reduce the
-type class burden, we instead only require a bornology on the underlying type, and a bounded
-set containing all the elements of the basis.
+An *approximate unit* is a filter `l` such that multiplication on the left (or right) by `m : α`
+tends to `𝓝 m` along the filter, and additionally `l ≠ ⊥` and `Disjoint l (cobounded α)`.
+
+In many cases of interest, an approximate unit is specified by a filter basis with certain
+properties, and the filter itself is of less interest. For instance, in the case of non-unital
+normed rings, it is often required that the elements of the basis are contained in the unit ball.
+In order to reduce the type class burden, we instead only require a bornology on the underlying
```

Comment:
I don't understand why we would need a basis for this. Can't we just say that `l ≤ principal (Metric.ball 0 1)`?

Reply:
Indeed you can there. I updated this to stop beating around the bush and just focus on the case I have in mind, which is obviously C⋆-algebras. I think the claim probably holds in other settings too (e.g., for approximate units in the convolution algebra $$L^1(\mathbb{R})$$).